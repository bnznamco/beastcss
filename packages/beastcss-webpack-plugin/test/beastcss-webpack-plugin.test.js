import { JSDOM } from 'jsdom';
import HtmlWebpackPlugin from 'html-webpack-plugin';
import MiniCssExtractPlugin from 'mini-css-extract-plugin';
import { readFile, compileToHtml, cleanDist } from './_helpers';
import BeastcssWebpackPlugin from '../src';

/**
 * Webpack configuration options
 *
 * @param {object} config webpack configuration
 */
function configure(config) {
  config.module.rules.push({
    test: /\.css$/,
    use: [MiniCssExtractPlugin.loader, 'css-loader'],
  });
  config.plugins.push(
    new MiniCssExtractPlugin({
      filename: '[name].css',
      chunkFilename: '[name].chunk.css',
    }),
    new HtmlWebpackPlugin({
      filename: 'index.html',
      template: 'index.html',
      inject: true,
      minify: false,
    })
  );
}

describe('Beastcss Webpack Plugin', () => {
  describe('webpack compilation', () => {
    let info;
    let html;

    beforeAll(async () => {
      ({ info, html } = await compileToHtml('basic', configure));
    });

    afterAll(async () => {
      await cleanDist('basic');
    });

    it('should have compiled 2 assets', () => {
      expect(info.assets).toHaveLength(2);
    });

    it('should match html snapshot', () => {
      expect(html).toMatchSnapshot();
    });
  });

  describe('Without html-webpack-plugin', () => {
    let html;
    let window;
    let document;

    beforeAll(async () => {
      ({ html } = await compileToHtml('raw', (config) => {
        config.module.rules.push(
          {
            test: /\.css$/,
            loader: 'css-loader',
          },
          {
            test: /\.html$/,
            loader: 'file-loader',
            options: {
              name: '[name].[ext]',
            },
          }
        );
      }));
      ({ window } = new JSDOM(html));
      ({ document } = window);
    });

    afterAll(async () => {
      await cleanDist('raw');
      if (window) {
        window.close();
      }
    });

    it('should insert used css to internal', () => {
      expect(document.getElementById('used')).not.toBeNull();
    });

    it('should omit unsed css', () => {
      expect(document.getElementById('unused')).toBeNull();
    });

    it('should match html snapshot', () => {
      expect(html).toMatchSnapshot();
    });
  });

  describe('additional stylesheets', () => {
    let html;
    let window;
    let document;

    beforeAll(async () => {
      ({ html } = await compileToHtml(
        'raw',
        (config) => {
          config.module.rules.push({
            test: /\.html|css$/,
            loader: 'file-loader',
            options: {
              name: '[name].[ext]',
            },
          });
        },
        {
          additionalStylesheets: ['**/additional.css'],
        }
      ));
      window = new JSDOM(html).window;
      document = window.document;
    });

    afterAll(async () => {
      await cleanDist('raw');
      if (window) {
        window.close();
      }
    });

    it('should insert used css to internal', () => {
      expect(document.getElementById('used')).not.toBeNull();
      expect(document.getElementById('used').textContent).toBe(
        'h1{color: green;}.additional{background: red;}'
      );
    });

    it('should omit unsed css', () => {
      expect(document.getElementById('unused')).toBeNull();
    });

    it('should match html snapshot', () => {
      expect(html).toMatchSnapshot();
    });
  });

  describe('Internal stylesheets', () => {
    afterAll(async () => {
      await cleanDist('basic');
      await cleanDist('unused');
    });

    it('should remove unused rules', async () => {
      const { html } = await compileToHtml('basic', configure);

      expect(html).not.toMatch(/\.extra-style/);
      expect(html).toMatchSnapshot();
    });

    it('should remove entire unused internal stylesheets', async () => {
      const { html } = await compileToHtml('unused', configure);
      const { window } = new JSDOM(html);
      const { document } = window;

      expect(document.querySelectorAll('style')).toHaveLength(1);
      expect(document.getElementById('unused')).toBeNull();
      expect(document.getElementById('used')).not.toBeNull();
      expect(document.getElementById('used').textContent).toMatchSnapshot();

      if (window) {
        window.close();
      }
    });
  });

  describe('External stylesheets', () => {
    let html;
    let window;

    beforeAll(async () => {
      ({ html } = await compileToHtml('external', configure, {
        pruneSource: true,
      }));
    });

    afterAll(async () => {
      await cleanDist('external');

      if (window) {
        window.close();
      }
    });

    it('should insert used css to internal', () => {
      expect(html).toMatch(/ul\.navbar\s*\{/);
    });

    it('should omit unsed css', () => {
      expect(html).not.toMatch(/\.extra-style/);
    });

    it('should load stylesheet async', () => {
      expect(html).toMatch(
        `<link href="main.css" rel="stylesheet" media="print" onload="this.media='all'; this.onload=null;">`
      );
    });

    it('should match html snapshot', () => {
      expect(html).toMatchSnapshot();
    });

    it('should prune external stylesheet', async () => {
      const externalCss = await readFile('fixtures/external/dist/main.css');

      expect(externalCss).toMatch(/\.extra-style\s*\{/);
      expect(externalCss).toMatchSnapshot();
    });
  });

  describe('publicPath', () => {
    let html;
    let window;
    let document;

    beforeAll(async () => {
      ({ html } = await compileToHtml(
        'external',
        (config) => {
          configure(config);
          config.output.publicPath = '/_public/';
        },
        {
          pruneSource: true,
        }
      ));
      ({ window } = new JSDOM(html));
      ({ document } = window);
    });

    afterAll(async () => {
      await cleanDist('external');
      if (window) {
        window.close();
      }
    });

    it('should insert used css to internal', () => {
      expect(html).toMatch(/ul\.navbar\s*\{/);
    });

    it('should omit unsed css', () => {
      expect(html).not.toMatch(/\.extra-style/);
    });

    it('should reference from publicPath', () => {
      const link = document.querySelector('link[rel="stylesheet"]');

      expect(link).not.toBeNull();
      expect(link).toHaveProperty('href');
      expect(link.getAttribute('href')).toBe('/_public/main.css');
    });

    it('should match html snapshot', () => {
      expect(html).toMatchSnapshot();
    });

    it('should prune external sheet', async () => {
      const externalCss = await readFile('fixtures/external/dist/main.css');

      expect(externalCss).toMatch(/\.extra-style\s*\{/);
      expect(externalCss).toMatchSnapshot();
    });
  });

  describe('preloadExternalStylesheets', () => {
    let html;
    let window;
    let document;

    beforeAll(async () => {
      ({ html } = await compileToHtml('external', configure, {
        preloadExternalStylesheets: true,
      }));
      ({ window } = new JSDOM(html));
      ({ document } = window);
    });

    afterAll(async () => {
      await cleanDist('external');
      if (window) {
        window.close();
      }
    });

    it('should omit unsed css', () => {
      expect(html).not.toMatch(/\.extra-style/);
    });

    it('should place a link rel="preload" in <head>', () => {
      const preload = document.querySelector('link[rel="preload"]');

      expect(preload).not.toBeNull();
      expect(preload).toHaveProperty('href');
      expect(preload.getAttribute('href')).toBe('main.css');
      expect(preload.parentNode).toBe(document.head);
    });

    it('should load stylesheet async', () => {
      expect(html).toMatch(
        `<link href="main.css" rel="stylesheet" media="print" onload="this.media='all'; this.onload=null;">`
      );
    });

    it('should match html snapshot', () => {
      expect(html).toMatchSnapshot();
    });
  });

  describe('externalThreshold', () => {
    let html;
    let window;
    let document;

    beforeAll(async () => {
      ({ html } = await compileToHtml('external', configure, {
        externalThreshold: 10000,
      }));
      ({ window } = new JSDOM(html));
      ({ document } = window);
    });

    afterAll(async () => cleanDist('external'));

    it('should fully insert stylesheets below the given size to internal', () => {
      expect(document.querySelectorAll('style')).toHaveLength(1);
      expect(html).toMatch(/\.extra-style/);
    });
  });

  describe('keyframes', () => {
    describe('by default', () => {
      let html;

      beforeAll(async () => {
        ({ html } = await compileToHtml('keyframes', configure));
      });

      afterAll(async () => {
        await cleanDist('keyframes');
      });

      it('should insert used @keyframes to internal', () => {
        expect(html).toMatch(/@keyframes present/);
      });

      it('should omit unsed @keyframes', () => {
        expect(html).not.toMatch(/@keyframes not-present/);
      });
    });

    describe("set to 'false'", () => {
      let html;

      beforeAll(async () => {
        ({ html } = await compileToHtml('keyframes', configure, {
          keyframes: false,
        }));
      });

      afterAll(async () => {
        await cleanDist('keyframes');
      });

      it('should not insert used @keyframes to internal', () => {
        expect(html).not.toMatch(/@keyframes present/);
      });

      it('should not insert unsed @keyframes to internal', () => {
        expect(html).not.toMatch(/@keyframes not-present/);
      });
    });
  });

  describe('fontFace', () => {
    describe('by default', () => {
      let html;

      beforeAll(async () => {
        ({ html } = await compileToHtml('font-face', configure));
        // console.log(html);
      });

      afterAll(async () => {
        await cleanDist('font-face');
      });

      it('should not insert used @font-face to internal', () => {
        expect(html).not.toMatch(/@font-face.*Present/);
      });

      it('should not insert unused @font-face', () => {
        expect(html).not.toMatch(/@font-face.*NotPresent/);
      });
    });

    describe("set to 'true'", () => {
      let html;

      beforeAll(async () => {
        ({ html } = await compileToHtml('font-face', configure, {
          fontFace: true,
        }));
      });

      afterAll(async () => {
        await cleanDist('font-face');
      });

      it('should insert used @font-face to internal', () => {
        expect(html).toMatch(/@font-face.*Present/);
      });

      it('should not insert unsed @font-face to internal', () => {
        expect(html).not.toMatch(/@font-face.*NotPresent/);
      });
    });
  });

  describe('prune source on multiple html', () => {
    let html;
    let html2;
    let css;

    beforeAll(async () => {
      ({ html } = await compileToHtml(
        'multiple',
        (config) => {
          configure(config);
          config.plugins.push(
            new HtmlWebpackPlugin({
              filename: 'index2.html',
              template: 'index2.html',
              inject: true,
              minify: false,
            })
          );
        },
        {
          pruneSource: true,
        }
      ));

      html2 = await readFile('fixtures/multiple/dist/index2.html');

      css = await readFile('fixtures/multiple/dist/main.css');
    });

    afterAll(async () => {
      await cleanDist('multiple');
    });

    it('should insert used css to internal', () => {
      expect(html).toMatch(/ul\.navbar\s*\{/);
      expect(html2).toMatch(/ul\.navbar\s*\{/);
    });

    it('should omit unsed css', () => {
      expect(html).not.toMatch(/\.extra-style/);
      expect(html2).not.toMatch(/\.extra-style/);
    });

    it('should match html snapshot', () => {
      expect(html).toMatchSnapshot();
    });

    it('should match html2 snapshot', () => {
      expect(html2).toMatchSnapshot('test');
    });

    it('should match stylesheet snapshot', () => {
      expect(css).toMatchSnapshot();
    });
  });

  describe('prune source on multiple html without html-webpack-plugin', () => {
    let html;
    let html2;
    let css;

    beforeAll(async () => {
      ({ html } = await compileToHtml(
        'rawMultiple',
        (config) => {
          config.module.rules.push({
            test: /\.html|css$/,
            loader: 'file-loader',
            options: {
              name: '[name].[ext]',
            },
          });
        },
        { pruneSource: true }
      ));

      html2 = await readFile('fixtures/rawMultiple/dist/index2.html');

      css = await readFile('fixtures/rawMultiple/dist/style.css');
    });

    afterAll(async () => {
      await cleanDist('rawMultiple');
    });

    it('should insert used css to internal', () => {
      expect(html).toMatch(/ul\.navbar\s*\{/);
      expect(html2).toMatch(/ul\.navbar\s*\{/);
    });

    it('should omit unsed css', () => {
      expect(html).not.toMatch(/\.extra-style/);
      expect(html2).not.toMatch(/\.extra-style/);
    });

    it('should match html snapshot', () => {
      expect(html).toMatchSnapshot();
    });

    it('should match html2 snapshot', () => {
      expect(html2).toMatchSnapshot('test');
    });

    it('should match stylesheet snapshot', () => {
      expect(css).toMatchSnapshot();
    });
  });

  describe('webpack logging', () => {
    let spyTrace;
    let spyDebug;
    let spyInfo;
    let spyWarn;
    let spyError;

    beforeAll(async () => {
      await compileToHtml(
        'basic',
        (config) => {
          configure(config);

          config.plugins.push({
            apply: (compiler) => {
              const getLogger = compiler.getInfrastructureLogger.bind(compiler);

              compiler.getInfrastructureLogger = jest.fn((name) => {
                const logger = getLogger(name);

                if (name === BeastcssWebpackPlugin.name) {
                  spyTrace = jest.spyOn(logger, 'trace').mockImplementation();
                  spyDebug = jest.spyOn(logger, 'debug').mockImplementation();
                  spyInfo = jest.spyOn(logger, 'info').mockImplementation();
                  spyWarn = jest.spyOn(logger, 'warn').mockImplementation();
                  spyError = jest.spyOn(logger, 'error').mockImplementation();
                }

                return logger;
              });
            },
          });
        },
        { logLevel: 'info' }
      );
    });

    afterAll(async () => {
      await cleanDist('basic');

      jest.restoreAllMocks();
    });

    it('should output logging with correct formatting', () => {
      expect(spyTrace).not.toHaveBeenCalled();
      expect(spyDebug).not.toHaveBeenCalled();
      expect(spyInfo).toHaveBeenCalled();
      expect(spyWarn).not.toHaveBeenCalled();
      expect(spyError).not.toHaveBeenCalled();
      expect(spyInfo.mock.calls[0][0]).toMatch(/^\[index\.html\] (.*)/);
    });
  });
});
