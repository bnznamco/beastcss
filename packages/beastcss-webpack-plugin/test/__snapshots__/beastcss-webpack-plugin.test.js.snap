// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Beastcss Webpack Plugin External stylesheets should match html snapshot 1`] = `
"<!DOCTYPE html>
<html>
  <head>
    <title>External CSS Demo</title>
  <script defer src=\\"bundle.js\\"></script><style>:root{font-size: 10px;}html{height: 100%;}body{padding-left: 11em;
  font-family: 'Times New Roman', times, serif;
  color: purple;
  background-color: #d8da3d;}*,:after,:before{box-sizing: inherit;}ul.navbar{list-style-type: none;
  padding: 0;
  margin: 0;
  position: absolute;
  top: 2em;
  left: 1em;
  width: 9em;}h1{font-family: helvetica, arial, sans-serif;}ul.navbar li{background: white;
  margin: 0.5em 0;
  padding: 0.3em;
  border-right: 1em solid black;}ul.navbar a{text-decoration: none;}a:link{color: blue;}a:visited{color: purple;}footer{margin-top: 1em;
  padding-top: 1em;
  border-top: thin dotted;}</style><link href=\\"main.css\\" rel=\\"stylesheet\\" media=\\"print\\" onload=\\"this.media='all'; this.onload=null;\\"><noscript><link rel=\\"stylesheet\\" href=\\"main.css\\"></noscript></head>
  <body>
    <ul class=\\"navbar\\">
      <li>
        <a href=\\"index.html\\">Home page</a>
      </li>
      <li>
        <a href=\\"musings.html\\">Musings</a>
      </li>
    </ul>
    <h1>My first styled page</h1>
    <p>Welcome to my styled page!</p>
    <footer>Made 5 April 2004</footer>
  </body>
</html>
"
`;

exports[`Beastcss Webpack Plugin External stylesheets should prune external stylesheet 1`] = `
":root {
  font-size: 10px;
}
html {
  height: 100%;
}
body {
  padding-left: 11em;
  font-family: 'Times New Roman', times, serif;
  color: purple;
  background-color: #d8da3d;
}
*,
:after,
:before {
  box-sizing: inherit;
}
ul.navbar {
  list-style-type: none;
  padding: 0;
  margin: 0;
  position: absolute;
  top: 2em;
  left: 1em;
  width: 9em;
}
h1 {
  font-family: helvetica, arial, sans-serif;
}
ul.navbar li {
  background: white;
  margin: 0.5em 0;
  padding: 0.3em;
  border-right: 1em solid black;
}
ul.navbar a {
  text-decoration: none;
}
a:link {
  color: blue;
}
a:visited {
  color: purple;
}
footer {
  margin-top: 1em;
  padding-top: 1em;
  border-top: thin dotted;
}
.extra-style {
  font-size: 200%;
}

"
`;

exports[`Beastcss Webpack Plugin Internal stylesheets should remove entire unused internal stylesheets 1`] = `"h1{color: green;}"`;

exports[`Beastcss Webpack Plugin Internal stylesheets should remove unused rules 1`] = `
"<!DOCTYPE html>
<html>
  <head>
    <title>Basic Demo</title>
    <style>body{padding-left: 11em;
        font-family: 'Times New Roman', times, serif;
        color: purple;
        background-color: #d8da3d;}ul.navbar{list-style-type: none;
        padding: 0;
        margin: 0;
        position: absolute;
        top: 2em;
        left: 1em;
        width: 9em;}h1{font-family: helvetica, arial, sans-serif;}ul.navbar li{background: white;
        margin: 0.5em 0;
        padding: 0.3em;
        border-right: 1em solid black;}ul.navbar a{text-decoration: none;}ul.navbar:not(.hidden) li:hover a{color: black;}ul.navbar:not(:hover) li:hover a{color: black;}ul.navbar li:hover a{color: red;}a:link{color: blue;}a:visited{color: purple;}footer{margin-top: 1em;
        padding-top: 1em;
        border-top: thin dotted;}.clazz\\\\:colon{color: green;}</style>
  <script defer src=\\"bundle.js\\"></script></head>
  <body>
    <ul class=\\"navbar clazz:colon\\">
      <li>
        <a href=\\"index.html\\">Home page</a>
      </li>
      <li>
        <a href=\\"musings.html\\">Musings</a>
      </li>
    </ul>
    <h1>My first styled page</h1>
    <p>Welcome to my styled page!</p>
    <footer>Made 5 April 2004</footer>
  </body>
</html>
"
`;

exports[`Beastcss Webpack Plugin Without html-webpack-plugin should match html snapshot 1`] = `
"<!DOCTYPE html>
<html>
  <head>
    <title>Basic Demo</title>
    
    <style id=\\"used\\">h1{color: green;}</style>
  </head>
  <body>
    <h1 class=\\"additional\\">Some HTML Here</h1>
  </body>
</html>
"
`;

exports[`Beastcss Webpack Plugin additional stylesheets should match html snapshot 1`] = `
"<!DOCTYPE html>
<html>
  <head>
    <title>Basic Demo</title>
    
    <style id=\\"used\\">h1{color: green;}.additional{background: red;}</style>
  </head>
  <body>
    <h1 class=\\"additional\\">Some HTML Here</h1>
  </body>
</html>
"
`;

exports[`Beastcss Webpack Plugin preloadExternalStylesheets should match html snapshot 1`] = `
"<!DOCTYPE html>
<html>
  <head>
    <title>External CSS Demo</title>
  <script defer src=\\"bundle.js\\"></script><style>:root{font-size: 10px;}html{height: 100%;}body{padding-left: 11em;
  font-family: 'Times New Roman', times, serif;
  color: purple;
  background-color: #d8da3d;}*,:after,:before{box-sizing: inherit;}ul.navbar{list-style-type: none;
  padding: 0;
  margin: 0;
  position: absolute;
  top: 2em;
  left: 1em;
  width: 9em;}h1{font-family: helvetica, arial, sans-serif;}ul.navbar li{background: white;
  margin: 0.5em 0;
  padding: 0.3em;
  border-right: 1em solid black;}ul.navbar a{text-decoration: none;}a:link{color: blue;}a:visited{color: purple;}footer{margin-top: 1em;
  padding-top: 1em;
  border-top: thin dotted;}</style><link rel=\\"preload\\" href=\\"main.css\\" as=\\"style\\"><link href=\\"main.css\\" rel=\\"stylesheet\\" media=\\"print\\" onload=\\"this.media='all'; this.onload=null;\\"><noscript><link rel=\\"stylesheet\\" href=\\"main.css\\"></noscript></head>
  <body>
    <ul class=\\"navbar\\">
      <li>
        <a href=\\"index.html\\">Home page</a>
      </li>
      <li>
        <a href=\\"musings.html\\">Musings</a>
      </li>
    </ul>
    <h1>My first styled page</h1>
    <p>Welcome to my styled page!</p>
    <footer>Made 5 April 2004</footer>
  </body>
</html>
"
`;

exports[`Beastcss Webpack Plugin prune source on multiple html should match html snapshot 1`] = `
"<!DOCTYPE html>
<html>
  <head>
    <title>External CSS Demo</title>
  <script defer src=\\"bundle.js\\"></script><style>:root{font-size: 10px;}html{height: 100%;}body{padding-left: 11em;
  font-family: 'Times New Roman', times, serif;
  color: purple;
  background-color: #d8da3d;}*,:after,:before{box-sizing: inherit;}ul.navbar{list-style-type: none;
  padding: 0;
  margin: 0;
  position: absolute;
  top: 2em;
  left: 1em;
  width: 9em;}h1{font-family: helvetica, arial, sans-serif;}ul.navbar li{background: white;
  margin: 0.5em 0;
  padding: 0.3em;
  border-right: 1em solid black;}ul.navbar a{text-decoration: none;}a:link{color: blue;}a:visited{color: purple;}footer{margin-top: 1em;
  padding-top: 1em;
  border-top: thin dotted;}:root{font-size: 10px;}html{height: 100%;}body{padding-left: 11em;
  font-family: 'Times New Roman', times, serif;
  color: purple;
  background-color: #d8da3d;}*,:after,:before{box-sizing: inherit;}ul.navbar{list-style-type: none;
  padding: 0;
  margin: 0;
  position: absolute;
  top: 2em;
  left: 1em;
  width: 9em;}h1{font-family: helvetica, arial, sans-serif;}ul.navbar li{background: white;
  margin: 0.5em 0;
  padding: 0.3em;
  border-right: 1em solid black;}ul.navbar a{text-decoration: none;}a:link{color: blue;}a:visited{color: purple;}footer{margin-top: 1em;
  padding-top: 1em;
  border-top: thin dotted;}</style><link href=\\"main.css\\" rel=\\"stylesheet\\" media=\\"print\\" onload=\\"this.media='print'; this.onload=null;\\"><noscript><link rel=\\"stylesheet\\" href=\\"main.css\\" media=\\"print\\"></noscript><noscript><link rel=\\"stylesheet\\" href=\\"main.css\\"></noscript></head>
  <body>
    <ul class=\\"navbar\\">
      <li>
        <a href=\\"index.html\\">Home page</a>
      </li>
      <li>
        <a href=\\"musings.html\\">Musings</a>
      </li>
    </ul>
    <h1>My first styled page</h1>
    <p>Welcome to my styled page!</p>
    <footer>Made 5 April 2004</footer>
  </body>
</html>
"
`;

exports[`Beastcss Webpack Plugin prune source on multiple html should match html2 snapshot: test 1`] = `
"<!DOCTYPE html>
<html>
  <head>
    <title>External CSS Demo</title>
  <script defer src=\\"bundle.js\\"></script><style>:root{font-size: 10px;}html{height: 100%;}body{padding-left: 11em;
  font-family: 'Times New Roman', times, serif;
  color: purple;
  background-color: #d8da3d;}*,:after,:before{box-sizing: inherit;}ul.navbar{list-style-type: none;
  padding: 0;
  margin: 0;
  position: absolute;
  top: 2em;
  left: 1em;
  width: 9em;}h1{font-family: helvetica, arial, sans-serif;}ul.navbar li{background: white;
  margin: 0.5em 0;
  padding: 0.3em;
  border-right: 1em solid black;}ul.navbar a{text-decoration: none;}a:link{color: blue;}a:visited{color: purple;}footer{margin-top: 1em;
  padding-top: 1em;
  border-top: thin dotted;}h2{color: red;}.bold{font-weight: bold;}:root{font-size: 10px;}html{height: 100%;}body{padding-left: 11em;
  font-family: 'Times New Roman', times, serif;
  color: purple;
  background-color: #d8da3d;}*,:after,:before{box-sizing: inherit;}ul.navbar{list-style-type: none;
  padding: 0;
  margin: 0;
  position: absolute;
  top: 2em;
  left: 1em;
  width: 9em;}h1{font-family: helvetica, arial, sans-serif;}ul.navbar li{background: white;
  margin: 0.5em 0;
  padding: 0.3em;
  border-right: 1em solid black;}ul.navbar a{text-decoration: none;}a:link{color: blue;}a:visited{color: purple;}footer{margin-top: 1em;
  padding-top: 1em;
  border-top: thin dotted;}h2{color: red;}.bold{font-weight: bold;}</style><link href=\\"main.css\\" rel=\\"stylesheet\\" media=\\"print\\" onload=\\"this.media='print'; this.onload=null;\\"><noscript><link rel=\\"stylesheet\\" href=\\"main.css\\" media=\\"print\\"></noscript><noscript><link rel=\\"stylesheet\\" href=\\"main.css\\"></noscript></head>
  <body>
    <ul class=\\"navbar\\">
      <li>
        <a href=\\"index.html\\">Home page</a>
      </li>
      <li>
        <a href=\\"musings.html\\">Musings</a>
      </li>
    </ul>
    <h1>My first styled page</h1>
    <p>Welcome to my styled page!</p>
    <h2>Heading</h2>
    <p class=\\"bold\\">bold paragraph</p>
    <footer>Made 5 April 2004</footer>
  </body>
</html>
"
`;

exports[`Beastcss Webpack Plugin prune source on multiple html should match stylesheet snapshot 1`] = `
":root {
  font-size: 10px;
}
html {
  height: 100%;
}
body {
  padding-left: 11em;
  font-family: 'Times New Roman', times, serif;
  color: purple;
  background-color: #d8da3d;
}
*,
:after,
:before {
  box-sizing: inherit;
}
ul.navbar {
  list-style-type: none;
  padding: 0;
  margin: 0;
  position: absolute;
  top: 2em;
  left: 1em;
  width: 9em;
}
h1 {
  font-family: helvetica, arial, sans-serif;
}
ul.navbar li {
  background: white;
  margin: 0.5em 0;
  padding: 0.3em;
  border-right: 1em solid black;
}
ul.navbar a {
  text-decoration: none;
}
a:link {
  color: blue;
}
a:visited {
  color: purple;
}
footer {
  margin-top: 1em;
  padding-top: 1em;
  border-top: thin dotted;
}
h2 {
  color: red;
}
.bold {
  font-weight: bold;
}
.extra-style {
  font-size: 200%;
}

"
`;

exports[`Beastcss Webpack Plugin prune source on multiple html without html-webpack-plugin should match html snapshot 1`] = `
"<!DOCTYPE html>
<html>
  <head>
    <title>External CSS Demo</title>
    <style>:root{font-size: 10px;}html{height: 100%;}body{padding-left: 11em;
  font-family: 'Times New Roman', times, serif;
  color: purple;
  background-color: #d8da3d;}*,:after,:before{box-sizing: inherit;}ul.navbar{list-style-type: none;
  padding: 0;
  margin: 0;
  position: absolute;
  top: 2em;
  left: 1em;
  width: 9em;}h1{font-family: helvetica, arial, sans-serif;}ul.navbar li{background: white;
  margin: 0.5em 0;
  padding: 0.3em;
  border-right: 1em solid black;}ul.navbar a{text-decoration: none;}a:link{color: blue;}a:visited{color: purple;}footer{margin-top: 1em;
  padding-top: 1em;
  border-top: thin dotted;}</style><link rel=\\"stylesheet\\" href=\\"./style.css\\" media=\\"print\\" onload=\\"this.media='all'; this.onload=null;\\"><noscript><link rel=\\"stylesheet\\" href=\\"./style.css\\"></noscript>
  </head>
  <body>
    <ul class=\\"navbar\\">
      <li>
        <a href=\\"index.html\\">Home page</a>
      </li>
      <li>
        <a href=\\"musings.html\\">Musings</a>
      </li>
    </ul>
    <h1>My first styled page</h1>
    <p>Welcome to my styled page!</p>
    <footer>Made 5 April 2004</footer>
  </body>
</html>
"
`;

exports[`Beastcss Webpack Plugin prune source on multiple html without html-webpack-plugin should match html2 snapshot: test 1`] = `
"<!DOCTYPE html>
<html>
  <head>
    <title>External CSS Demo</title>
    <style>:root{font-size: 10px;}html{height: 100%;}body{padding-left: 11em;
  font-family: 'Times New Roman', times, serif;
  color: purple;
  background-color: #d8da3d;}*,:after,:before{box-sizing: inherit;}ul.navbar{list-style-type: none;
  padding: 0;
  margin: 0;
  position: absolute;
  top: 2em;
  left: 1em;
  width: 9em;}h1{font-family: helvetica, arial, sans-serif;}ul.navbar li{background: white;
  margin: 0.5em 0;
  padding: 0.3em;
  border-right: 1em solid black;}ul.navbar a{text-decoration: none;}a:link{color: blue;}a:visited{color: purple;}footer{margin-top: 1em;
  padding-top: 1em;
  border-top: thin dotted;}h2{color: red;}.bold{font-weight: bold;}</style><link rel=\\"stylesheet\\" href=\\"./style.css\\" media=\\"print\\" onload=\\"this.media='all'; this.onload=null;\\"><noscript><link rel=\\"stylesheet\\" href=\\"./style.css\\"></noscript>
  </head>
  <body>
    <ul class=\\"navbar\\">
      <li>
        <a href=\\"index.html\\">Home page</a>
      </li>
      <li>
        <a href=\\"musings.html\\">Musings</a>
      </li>
    </ul>
    <h1>My first styled page</h1>
    <p>Welcome to my styled page!</p>
    <h2>Heading</h2>
    <p class=\\"bold\\">bold paragraph</p>
    <footer>Made 5 April 2004</footer>
  </body>
</html>
"
`;

exports[`Beastcss Webpack Plugin prune source on multiple html without html-webpack-plugin should match stylesheet snapshot 1`] = `".extra-style{font-size: 200%;}"`;

exports[`Beastcss Webpack Plugin publicPath should match html snapshot 1`] = `
"<!DOCTYPE html>
<html>
  <head>
    <title>External CSS Demo</title>
  <script defer src=\\"/_public/bundle.js\\"></script><style>:root{font-size: 10px;}html{height: 100%;}body{padding-left: 11em;
  font-family: 'Times New Roman', times, serif;
  color: purple;
  background-color: #d8da3d;}*,:after,:before{box-sizing: inherit;}ul.navbar{list-style-type: none;
  padding: 0;
  margin: 0;
  position: absolute;
  top: 2em;
  left: 1em;
  width: 9em;}h1{font-family: helvetica, arial, sans-serif;}ul.navbar li{background: white;
  margin: 0.5em 0;
  padding: 0.3em;
  border-right: 1em solid black;}ul.navbar a{text-decoration: none;}a:link{color: blue;}a:visited{color: purple;}footer{margin-top: 1em;
  padding-top: 1em;
  border-top: thin dotted;}</style><link href=\\"/_public/main.css\\" rel=\\"stylesheet\\" media=\\"print\\" onload=\\"this.media='all'; this.onload=null;\\"><noscript><link rel=\\"stylesheet\\" href=\\"/_public/main.css\\"></noscript></head>
  <body>
    <ul class=\\"navbar\\">
      <li>
        <a href=\\"index.html\\">Home page</a>
      </li>
      <li>
        <a href=\\"musings.html\\">Musings</a>
      </li>
    </ul>
    <h1>My first styled page</h1>
    <p>Welcome to my styled page!</p>
    <footer>Made 5 April 2004</footer>
  </body>
</html>
"
`;

exports[`Beastcss Webpack Plugin publicPath should prune external sheet 1`] = `
":root {
  font-size: 10px;
}
html {
  height: 100%;
}
body {
  padding-left: 11em;
  font-family: 'Times New Roman', times, serif;
  color: purple;
  background-color: #d8da3d;
}
*,
:after,
:before {
  box-sizing: inherit;
}
ul.navbar {
  list-style-type: none;
  padding: 0;
  margin: 0;
  position: absolute;
  top: 2em;
  left: 1em;
  width: 9em;
}
h1 {
  font-family: helvetica, arial, sans-serif;
}
ul.navbar li {
  background: white;
  margin: 0.5em 0;
  padding: 0.3em;
  border-right: 1em solid black;
}
ul.navbar a {
  text-decoration: none;
}
a:link {
  color: blue;
}
a:visited {
  color: purple;
}
footer {
  margin-top: 1em;
  padding-top: 1em;
  border-top: thin dotted;
}
.extra-style {
  font-size: 200%;
}

"
`;

exports[`Beastcss Webpack Plugin webpack compilation should match html snapshot 1`] = `
"<!DOCTYPE html>
<html>
  <head>
    <title>Basic Demo</title>
    <style>body{padding-left: 11em;
        font-family: 'Times New Roman', times, serif;
        color: purple;
        background-color: #d8da3d;}ul.navbar{list-style-type: none;
        padding: 0;
        margin: 0;
        position: absolute;
        top: 2em;
        left: 1em;
        width: 9em;}h1{font-family: helvetica, arial, sans-serif;}ul.navbar li{background: white;
        margin: 0.5em 0;
        padding: 0.3em;
        border-right: 1em solid black;}ul.navbar a{text-decoration: none;}ul.navbar:not(.hidden) li:hover a{color: black;}ul.navbar:not(:hover) li:hover a{color: black;}ul.navbar li:hover a{color: red;}a:link{color: blue;}a:visited{color: purple;}footer{margin-top: 1em;
        padding-top: 1em;
        border-top: thin dotted;}.clazz\\\\:colon{color: green;}</style>
  <script defer src=\\"bundle.js\\"></script></head>
  <body>
    <ul class=\\"navbar clazz:colon\\">
      <li>
        <a href=\\"index.html\\">Home page</a>
      </li>
      <li>
        <a href=\\"musings.html\\">Musings</a>
      </li>
    </ul>
    <h1>My first styled page</h1>
    <p>Welcome to my styled page!</p>
    <footer>Made 5 April 2004</footer>
  </body>
</html>
"
`;
